cli/azd/cmd/deploy.go

```diff
@@ -235,14 +225,30 @@ func (da *deployAction) Run(ctx context.Context) (*actions.ActionResult, error)
 			continue
 		}
 
-		var packageOutput *project.ServicePackageResult
+		var packageResult *project.ServicePackageResult
 		if da.flags.fromPackage != "" {
-			packageOutput = &project.ServicePackageResult{
+			// --from-package set, skip packaging
+			packageResult = &project.ServicePackageResult{
 				PackagePath: da.flags.fromPackage,
 			}
+		} else {
+			//  --from-package not set, package the application
+			packageTask := da.serviceManager.Package(ctx, svc, nil)
+			go func() {
+				for packageProgress := range packageTask.Progress() {
+					progressMessage := fmt.Sprintf("Deploying service %s (%s)", svc.Name, packageProgress.Message)
+					da.console.ShowSpinner(ctx, progressMessage, input.Step)
+				}
+			}()
+
+			packageResult, err = packageTask.Await()
+			if err != nil {
+				da.console.StopSpinner(ctx, stepMessage, input.StepFailed)
+				return nil, err
+			}
```

Comment:
I don't really agree with the direction of not showing the `package` cmd output vs calling the `serviceManager` but this will work fine.

I'd propose that we add an issue to introduce an internal `quiet` flag or similar so that we can call the command internally so that we do not need to manage packaging the application within multiple commands/actions.

Reply:
https://github.com/Azure/azure-dev-pr/issues/1549